extensions/ql-vscode/src/variant-analysis/store/repo-states-store.ts

```diff
@@ -0,0 +1,17 @@
+import { outputJson, readJson } from "fs-extra";
+import { VariantAnalysisScannedRepositoryState } from "../shared/variant-analysis";
+
+export const REPO_STATES_FILENAME = "repo_states.json";
+
+export async function writeRepoStates(
+  storagePath: string,
+  repoStates: Record<number, VariantAnalysisScannedRepositoryState> | undefined,
```

Comment:
Hmm should we allow this to be `undefined`? Is it what the current behaviour is doing?

Reply:
yes, it seems to mirror current behaviour (TS complained without the undefined). I agree however that we should find out where that comes from and if we can remove it when working on the next steps.